widget/tabcontainer.go

```diff
@@ -380,8 +381,25 @@ func (r *tabContainerRenderer) moveSelection() {
 		underlinePos = fyne.NewPos(r.tabBar.Position().X-theme.Padding(), selected.Position().Y)
 		underlineSize = fyne.NewSize(theme.Padding(), selected.Size().Height)
 	}
-	r.underline.Resize(underlineSize)
-	r.underline.Move(underlinePos)
+
+	if r.underline.Position().IsZero() || r.underline.Position() == underlinePos {
+		r.underline.Move(underlinePos)
+		r.underline.Resize(underlineSize)
+	} else if r.animation == nil {
+		r.animation = canvas.NewPositionAnimation(r.underline.Position(), underlinePos, canvas.DurationStandard, func(p fyne.Position) {
+			r.underline.Move(p)
+			canvas.Refresh(r.underline)
+			if p == underlinePos {
+				r.animation = nil
+			}
+		})
+		r.animation.Start()
+
+		canvas.NewSizeAnimation(r.underline.Size(), underlineSize, canvas.DurationStandard, func(s fyne.Size) {
```

Comment:
```suggestion
		canvas.NewSizeAnimation(r.underline.Size(), underlineSize, time.Millisecond * 100, func(s fyne.Size) {
```
I did some testing and 100 milliseconds is what I would consider the best.